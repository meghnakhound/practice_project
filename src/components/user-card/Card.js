import React, { Component } from "react";
import "./Card.css";

class Card extends Component {
  constructor(props) {
    super(props);
    this.state = {
      userId: "",
      name: "",
      userImage: ""
    };
  }

  checkUserID = e => {
    const userId = e.target.value;
    this.setState({ userId: parseFloat(userId) });
  };

  fetchUserDetail = () => {
    const { userId } = this.state;
    if (userId) {
      fetch(`https://reqres.in/api/users/${userId}`)
        .then(res => res.json())
        .then(result => {
          if (result.data) {
            this.setState({
              name: `${result.data.first_name} ${result.data.last_name}`,
              userImage: result.data.avatar
            });
          } else {
            this.setState({ userId: "", name: "", userImage: "" });
          }
        });
    } else {
      alert("Please enter a userId");
    }
  };

  render() {
    return (
      <div className="main-card-container">
        <input
          type="number"
          className="txtfield-userid"
          onChange={this.checkUserID}
          value={this.state.userId}
        />
        <button
          text="Get Details"
          className="btn-get-details"
          onClick={this.fetchUserDetail}
        />
        <div className="card">
          <div className="image-container">
            <img src={this.state.userImage} alt={this.state.name} />
          </div>
          <div className="details-container">
            <span>
              ID: {this.state.userImage && <span>{this.state.userId}</span>}
            </span>
            <span>Name: {this.state.name}</span>
          </div>
        </div>
      </div>
    );
  }
}

export default Card;
