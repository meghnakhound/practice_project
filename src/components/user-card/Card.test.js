import React, {Component} from 'react';
import {shallow} from 'enzyme';
import renderer from 'react-test-renderer';
import Card from './Card';

test('Button component renders correctly', () => {
  const component = renderer
    .create(<Card/>)
    .toJSON();
  expect(component).toMatchSnapshot()
});
