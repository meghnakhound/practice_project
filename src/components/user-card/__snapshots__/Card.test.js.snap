// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component renders correctly 1`] = `
<div
  className="main-card-container"
>
  <input
    className="txtfield-userid"
    onChange={[Function]}
    type="text"
    value=""
  />
  <button
    className="btn-get-details"
    onClick={[Function]}
    text="Get Details"
  />
  <div
    className="card"
  >
    <div
      className="image-container"
    >
      <img
        src=""
      />
    </div>
    <div
      className="details-container"
    >
      <span>
        ID: 
        
      </span>
      <span>
        Name: 
        
      </span>
    </div>
  </div>
</div>
`;
